Properly rename shader config on update/version change

TheKodeToad opened the pull request:
Parent PR: #1588

~~Would appreciate suggestions on how to do this better :>~~
Managed to remove the need for m_targetPath by removing some logic that's unused anyway since cauldron loader doesn't apply to shader packs (it was accidentally changed from canvas) nor seem to exist on CurseForge or modrinth any more
This logic could be fixed, but it's probably best to take a different approach such as showing things which should go to the `resourcepacks` directory inside the resource pack downloader dialog.
Also for some reason the warning doesn't show if the renaming fails

TheKodeToad committed:
Properly rename shader config on update/version change

Signed-off-by: TheKodeToad <TheKodeToad@proton.me>

Trial97 commented on launcher/ResourceDownloadTask.cpp:
```diff
@@ -91,8 +95,25 @@ void ResourceDownloadTask::downloadSucceeded()
+        if (config.exists()) {
+            QString src = config.filePath();
+            QString dest = m_targetPath + ".txt";
+            bool success = QFile::rename(src, dest);
```
Consider using `FS::move` as this one ensures that if the rename fails, the dest is still created by copy(even if for some reasons the src can't be renamed or deleted; Windows issues with open files)

TheKodeToad replied:
I'm not a big fan of how FS::move overwrites the destination... I suppose I can add an extra check for that

TheKodeToad committed:
Use FS::move instead; check new config does not exist

Signed-off-by: TheKodeToad <TheKodeToad@proton.me>

TheKodeToad committed:
Simplify implementation by removing some unused code

Only applies to cauldron loader which doesn't exist on Modrinth or
CurseForge, and doesn't even make sense for shaders.

Signed-off-by: TheKodeToad <TheKodeToad@proton.me>
